core/src/test/scala/integration/kafka/api/AuthorizerIntegrationTest.scala

```diff
@@ -405,6 +408,34 @@ class AuthorizerIntegrationTest extends KafkaServerTestHarness {
   }
 
   @Test
+  def testPatternSubscriptionWithNoTopicAccess() {
+    addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Write)), topicResource)
+    sendRecords(1, tp)
+    removeAllAcls()
+
+    addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Read)), groupResource)
+    try {
+      this.consumers.head.subscribe(Pattern.compile(topicPattern), new NoOpConsumerRebalanceListener)
+      consumeRecords(this.consumers.head)
```

Comment:
I changed the implementation in the following way and I didn't get a `TopicAuthorizationException`:

```java
                    Set<String> unauthorizedTopics = new HashSet<String>();
//                    for (String topic : cluster.unauthorizedTopics()) {
//                        if (subscriptions.getSubscribedPattern().matcher(topic).matches())
//                            unauthorizedTopics.add(topic);
//                    }
                    if (!unauthorizedTopics.isEmpty())
                        throw new TopicAuthorizationException(unauthorizedTopics);
```

Instead I got a "java.lang.IllegalStateException: Failed to consume the expected records after 51 iterations".